### PR TITLE
Simplified code around running rewriters and conditions

### PR DESCRIPTION
**Pull request by davidegrohmann**

### CODE REVIEW

File: community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/tracing/rewriters/RewriterTaskBuilder.scala [hunk 1]
```diff
@@ -21,44 +21,21 @@ package org.neo4j.cypher.internal.compiler.v2_2.tracing.rewriters
...
-
-    private def withConditionsAppended =
-      if (conditions.isEmpty) self else copy(tasks = tasks :+ RunConditions(previousName, conditions))
+    def +(condition: RewriterCondition) = copy(conditions = conditions + condition)
+    def -(condition: RewriterCondition) = copy(conditions = conditions - condition)
+    def allTasks = if (conditions.isEmpty) tasks else tasks :+ RunConditions(previousName, conditions)
   }
+
+  def apply(steps: Seq[RewriterStep]): Seq[RewriterTask] = steps.foldLeft(State()) {
+    case (state, ApplyRewriter(name, rewriter)) => state +(name, rewriter)
```
Comment (pontusmelke) [inline community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/tracing/rewriters/RewriterTaskBuilder.scala]: Not sure I am fully following but shouldn't we add `copy(tasks = tasks :+ RunConditions(previousName, conditions))` as is done in `withConditionsAppended` in the old code?

Comment (davidegrohmann) [inline community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/tracing/rewriters/RewriterTaskBuilder.scala]: Hmm, I see what you mean, probably you are right. I'll add a test and fix it.

Comment (davidegrohmann) [inline community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/tracing/rewriters/RewriterTaskBuilder.scala]: @pontusmelke Fixed

**Comment by benbc** [comment]
@davidegrohmann The build for this PR has actually passed, but didn't update the status here properly. I have triggered another build run, which will turn this green, but if it's urgent to get this merged then you could consider it passed already.

**Pull request merged**